spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/rdd/partitioner/ReplicaPartitioner.scala

```diff
@@ -0,0 +1,34 @@
+package com.datastax.spark.connector.rdd.partitioner
+
+import java.net.InetAddress
+
+import com.datastax.spark.connector.cql.CassandraConnector
+import org.apache.spark.{Partition, Partitioner}
+
+
+case class EndpointPartition(index: Int, endpoint: Set[InetAddress]) extends Partition
+
+class ReplicaPartitioner(partitionsPerReplicaSet: Int, connector:CassandraConnector) extends Partitioner {
+    val hosts = connector.hosts
+    val hostMap = hosts.zipWithIndex.toMap
+    /*TODO We Need JAVA-312 to get sets of replicas instead of single endpoints. Once we have that we'll be able to
+    build a map of Set[ip,ip,...] => Index before looking at our data. */
+    val indexMap = (0 to partitionsPerReplicaSet).flatMap(offset => hostMap.map { case (hosts, index) => (index + offset, Set(hosts))}).toMap
```

Comment:
in `case (hosts, index)` you really should use different name for `hosts` because (1) it masks `hosts` value and (2) it is really a single host.